src/ints/int10_vesa.cpp

```diff
@@ -91,48 +91,73 @@ struct MODE_INFO{
 #pragma pack()
 #endif
 
+uint8_t VESA_GetSVGAInformation(const uint16_t segment, const uint16_t offset)
+{
+	// Fill buffer with VESA information
+	PhysPt buffer = PhysicalMake(segment, offset);
+	const auto id = mem_readd(buffer);
 
+	const auto vbe2 = (((id == 0x56424532) || (id == 0x32454256)) &&
+	                   !int10.vesa_oldvbe);
 
-uint8_t VESA_GetSVGAInformation(uint16_t seg,uint16_t off) {
-	/* Fill 256 byte buffer with VESA information */
-	PhysPt buffer=PhysicalMake(seg,off);
-	Bitu i;
-	bool vbe2=false;uint16_t vbe2_pos=256+off;
-	Bitu id=mem_readd(buffer);
-	if (((id==0x56424532)||(id==0x32454256)) && (!int10.vesa_oldvbe)) vbe2=true;
-	if (vbe2) {
-		for (i=0;i<0x200;i++) mem_writeb(buffer+i,0);		
-	} else {
-		for (i=0;i<0x100;i++) mem_writeb(buffer+i,0);
+	const auto vbe_bufsize = vbe2 ? 0x200 : 0x100;
+	for (auto i = 0; i < vbe_bufsize; i++) {
+		mem_writeb(buffer + i, 0);
 	}
-	/* Fill common data */
-	MEM_BlockWrite(buffer,(void *)"VESA",4);				//Identification
-	if (!int10.vesa_oldvbe) mem_writew(buffer+0x04,0x200);	//Vesa version 2.0
-	else mem_writew(buffer+0x04,0x102);						//Vesa version 1.2
+
+	// Identification
+	auto vesa_string = "VESA";
+	MEM_BlockWrite(buffer, const_cast<char*>(vesa_string), 4);
```

Comment:
`const_cast<char*>` -> `static_cast<const void*>` - this should get rid of one PVS Studio warning.